azurelinuxagent/common/osutil/default.py

```diff
@@ -1420,3 +1424,37 @@ def _update_nic_state(self, state, ip_command, handler, description):
                     handler(state[interface_name], result.group(2))
                 else:
                     logger.error("Interface {0} has {1} but no link state".format(interface_name, description))
+
+    def supports_logrotate(self):
+        """
+        logrotate is allows automatic rotation, compression, removal, and mailing of log files.
+
+        :return: bool: Does the OS support logrotate
+        """
+        return True
```

Comment:
> Do most distros (or at least the ones we endorse) support `logrotate`? Have we tested this?

The log rotation behavior currently used is `logrotate` and I didn't want to change that without data. So kept the Default as true, and la

> If not, it would be better to set the default behavior to "not supported" and "support" in respective OS utils classes.

This could be done later, once we figure out the state of logrotate. But anecdotally, most endorsed distros support `logrotate` and very few special cases don't have logrotate available.

Reply:
So, if a distro doesn't support `logrotate`, it will inherit `True` from the default OS util and no logs will be rotated since the manual logic for log rotation won't be invoked. Since the purpose of this PR is to add log rotation, this means we can potentially leave **endorsed distros** without log rotation still.

I think it's worth investigating which of the endorsed distros don't support `logrotate` (if any) and override their `supports_logrotate` method in order to invoke manual log rotation, if needed.